src/field_theory/primitive_element.lean

```diff
@@ -0,0 +1,260 @@
+  { intro eq_zero,
+    rw [polynomial.leading_coeff_eq_zero,euclidean_domain.gcd_eq_zero_iff] at eq_zero,
+    apply polynomial.map_monic_ne_zero (minimal_polynomial.monic hβ) eq_zero.2, },
+  have finale : β = algebra_map F⟮γ⟯ E (-p.coeff 0 / p.coeff 1),
+  { rw [ring_hom.map_div,ring_hom.map_neg,←polynomial.coeff_map,←polynomial.coeff_map,←hswap,
+        polynomial.coeff_C_mul,polynomial.coeff_C_mul,polynomial.coeff_sub,polynomial.coeff_sub,
+        polynomial.coeff_X_zero,polynomial.coeff_X_one,polynomial.coeff_C_zero,polynomial.coeff_C,
+        zero_sub,mul_neg_eq_neg_mul_symm,neg_neg],
+    simp only [one_ne_zero],
+    rw [if_false,sub_zero,mul_one,mul_comm,div_eq_mul_inv,mul_assoc,
+        mul_inv_cancel leading_coeff_ne_zero,mul_one] },
+  have β_in_Fγ : β ∈ F⟮γ⟯,
+  { rw finale,
+    exact subtype.mem (-p.coeff 0 / p.coeff 1) },
+  have γ_in_Fγ : γ ∈ F⟮γ⟯ := field.mem_adjoin_simple_self F γ,
+  have c_in_Fγ : ιFE c ∈ F⟮γ⟯ := field.adjoin.algebra_map_mem F {γ} c,
+  have cβ_in_Fγ : (ιFE c)*β ∈ (F⟮γ⟯ : set E) := is_submonoid.mul_mem c_in_Fγ β_in_Fγ,
+  have α_in_Fγ : α ∈ (F⟮γ⟯ : set E),
+  { rw (show α = γ - (ιFE c)*β, by exact (add_sub_cancel α ((ιFE c)*β)).symm),
+    exact is_add_subgroup.sub_mem F⟮γ⟯ γ ((ιFE c)*β) γ_in_Fγ cβ_in_Fγ, },
+  have αβ_in_Fγ : {α,β} ⊆ (F⟮γ⟯ : set E) := λ x hx, by cases hx; cases hx; cases hx; assumption,
+  have Fαβ_sub_Fγ : (F⟮α, β⟯ : set E) ⊆ (F⟮γ⟯ : set E) :=
+    (field.adjoin_subset_iff F {α,β}).mp αβ_in_Fγ,
+  exact ⟨γ, Fαβ_sub_Fγ⟩,
+end
+
+section primitive_element_same_universe
+universe u
+
+/-- Primitive element theorem for infinite fields. -/
+theorem primitive_element_inf {F E : Type u} [field F] [field E] [algebra F E]
+  (F_sep : is_separable F E) (F_findim: finite_dimensional F E) (F_inf : infinite F)
+  (n : ℕ) (hn : findim F E = n) :
+  ∃ α : E, F⟮α⟯ = ⊤ :=
+begin
+  tactic.unfreeze_local_instances,
+  revert F,
+  apply nat.strong_induction_on n,
+  clear n,
+  rintros n ih F hF hFE F_sep F_findim F_inf rfl,
+  by_cases key : ∃ α : E, findim F F⟮α⟯ > 1,
+  { cases key with α hα,
+    haveI Fα_findim : finite_dimensional F⟮α⟯ E := finite_dimensional.right F F⟮α⟯ E,
+    have Fα_dim_lt_F_dim : findim F⟮α⟯ E < findim F E,
+    { rw ← findim_mul_findim F F⟮α⟯ E,
+      nlinarith [show 0 < findim F⟮α⟯ E, from findim_pos, show 0 < findim F F⟮α⟯, from findim_pos], },
+    have Fα_inf : infinite F⟮α⟯ := infinite.of_injective _ (algebra_map F F⟮α⟯).injective,
+    have Fα_sep : is_separable F⟮α⟯ E := is_separable_tower_top_of_is_separable_tower F F⟮α⟯ E F_sep,
+    obtain ⟨β, hβ⟩ := ih _ Fα_dim_lt_F_dim Fα_sep Fα_findim Fα_inf rfl,
+    obtain ⟨γ, hγ⟩ := primitive_element_two_inf α β F_sep F_inf,
+    simp only [←adjoin_simple_adjoin_simple, subalgebra.ext_iff, algebra.mem_top, iff_true, *] at *,
+    exact ⟨γ, λ x, hγ algebra.mem_top⟩, },
+  { push_neg at key,
+    rw ← bot_eq_top_of_findim_adjoin_le_one key,
+    exact ⟨0, by rw adjoin_zero⟩, },
+end
```

Comment:
(Using `intermediate_field` would also fix the problems with new universes appearing in the induction.)

Reply:
Yeah, I think `intermediate_field` will make a lot of things nicer. Once that PR gets merged we'll work on converting this stuff to use it (along with bundled subfields, etc).

Reply:
I'm going to mark this as "resolved" because we'll deal with it in a future PR after `intermediate_field` has been merged. Feel free to "unresolve" it if you think there's something more we should do now.